scripts/hashlint.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
+
+"""
+Simple script to validate the various hashes for downloads across the project.
+"""
+
+import hashlib
+import sys
+
+import requests
+import yaml
+
+
+URLS = {
+    "roles/jgrasp/vars/main.yml": {
+        "hash": "jgrasp.hash",
+        "urls": ["jgrasp.url"],
+    },
+    "roles/finch/vars/main.yml": {
+        "hash": "finch.hash",
+        "urls": ["finch.url"],
+    },
+    "roles/eclipse/vars/main.yml": {
+        "hash": "eclipse.hash",
+        "urls": ["eclipse.url", "eclipse.url_backup"],
+    },
+}
+
+
+def get_field(data, key):
+    """
+    Get a field separated by dots
+    """
+
+    if isinstance(key, str):
+        key = key.split('.')
+    else:
+        print(type(key))
+    while key:
+        data = data[key.pop(0)]
+    return data
+
+
+def check_software_hash(session, url, expected, file):
+    """
+    Checks the hash for the contents of a URL against an expected value
+    """
+
+    try:
+        data = session.get(url).content
+    # pylint: disable=broad-except
+    except Exception:
+        print(f"{file}: Unable to download {url}", file=sys.stderr)
+        return False
+
+    sha1 = hashlib.sha1(data).hexdigest()
+    if sha1 != expected:
+        print(f"{file}: Expected {expected}. Found {sha1}", file=sys.stderr)
+        return False
+
+    return True
+
+
+def main():
+    """
+    Main
+    """
+
+    session = requests.Session()
```

Comment:
I was going to ask if there's a reason to recycle session between requests, but since we're not using auth or cookies, is a session even necessary?

Reply:
If we apply the suggestion for the 'User-Agent' header, we can do it on the session instead of each request. A session isn't really required though.